google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/GenerateHistoricalMetrics.java

```diff
@@ -0,0 +1,157 @@
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Runs the code example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param planId the plan ID.
+   */
+  // [START generate_historical_metrics]
+  private void runExample(GoogleAdsClient googleAdsClient, Long customerId, Long planId) {
+    String resourceName = ResourceNames.keywordPlan(customerId, planId);
+
+    try (KeywordPlanServiceClient keywordPlanServiceClient =
+        googleAdsClient.getLatestVersion().createKeywordPlanServiceClient()) {
+      GenerateHistoricalMetricsResponse response =
+          keywordPlanServiceClient.generateHistoricalMetrics(resourceName);
+
+      for (KeywordPlanKeywordHistoricalMetrics metric : response.getMetricsList()) {
+        // These metrics include those for both the search query and any close variants included in
+        // the response.
+
+        ProtocolStringList variants = metric.getCloseVariantsList();
+        String variantsList =
+            (variants.isEmpty())
+                ? ""
+                : String.format(" (and the following variants: %s)", String.join(", ", variants));
+
+        System.out.printf(
+            "The search query '%s'%s generated the following historical metrics:%n",
+            metric.getSearchQuery(), variantsList);
+
+        // Approximate number of monthly searches on this query averaged for the past 12 months.
+        System.out.printf(
+            "\tApproximate monthly searches: %d.%n",
+            metric.getKeywordMetrics().getAvgMonthlySearches());
+
+        // The competition level for this search query.
+        System.out.printf(
+            "\tCompetition level: %s.%n", metric.getKeywordMetrics().getCompetition().name());
+
+        // The competition index for the query in the range [0, 100]. This shows how competitive ad
+        // placement is for a keyword. The level of competition from 0-100 is determined by the
+        // number of ad slots filled divided by the total number of ad slots available. If not
+        // enough data is available, None will be returned.
+        System.out.printf(
+            "\tCompetition index: %d.%n", metric.getKeywordMetrics().getCompetitionIndex());
```

Comment:
I'd either:

a) conditionally print this based on `hasCompetitionIndex()` or
b) print "none" or "n/a" if not present (this'll require changing from `%d` to `%s`)

since getCompetitionIndex() will just return `0` if the metrics object doesn't have a value for the index.

Reply:
Same applies to lowTopOfPageBidMicros and highTopOfPageBidMicros if those can be `null`.